dspace-api/src/main/java/org/dspace/scripts/handler/impl/CommandLineDSpaceRunnableHandler.java

```diff
@@ -10,34 +10,115 @@
+import org.dspace.services.factory.DSpaceServicesFactory;
 
 /**
  * This is an implementation for the CommandLineDSpaceRunnables which means that these implementations
  * are used by DSpaceRunnables which are called from the CommandLine
  */
 public class CommandLineDSpaceRunnableHandler implements DSpaceRunnableHandler {
-    private static final Logger log = org.apache.logging.log4j.LogManager
+    private static final Logger log = LogManager
         .getLogger(CommandLineDSpaceRunnableHandler.class);
+    private final ConfigurationService configurationService =
+        DSpaceServicesFactory.getInstance().getConfigurationService();
+    private final boolean IS_SAVE_ENABLED = isSaveEnabled();
+    private final ProcessService processService = ScriptServiceFactory.getInstance().getProcessService();
+    private final EPersonService ePersonService = EPersonServiceFactory.getInstance().getEPersonService();
+    String scriptName;
+    Integer processId;
+    UUID ePersonUUID;
+
+    public CommandLineDSpaceRunnableHandler() {
+    }
+
+    public CommandLineDSpaceRunnableHandler(String scriptName, List<DSpaceCommandLineParameter> parameters) {
+        if (IS_SAVE_ENABLED) {
+            Context context = new Context();
+            try {
+                EPerson ePerson = null;
+                String parameter = parameters.get(0).getName();
+                if (parameter.contains("-e")) {
+                    String email = Arrays.stream(parameter.split(" "))
+                        .filter(s -> s.contains("@")).findFirst()
+                        .orElseThrow(() -> new IllegalArgumentException("No email found in parameters"));
+                    ePerson = ePersonService.findByEmail(context, email);
+                    if (ePerson == null) {
+                        throw new IllegalArgumentException("No eperson found with email: " + email);
+                    }
+                    this.ePersonUUID = ePerson.getID();
+                }
+                Process process = processService.create(context, ePerson, scriptName, parameters,
+                    new HashSet<>(context.getSpecialGroups()));
+                processId = process.getID();
+                this.scriptName = process.getName();
+                context.complete();
+            } catch (Exception e) {
+                logError("CommandLineDspaceRunnableHandler with ePerson: " + ePersonUUID + " for Script with name: " +
+                    scriptName +
+                    " and parameters: " + parameters + " could not be created", e);
+            } finally {
+                context.close();
+            }
+        }
+    }
+
 
     @Override
     public void start() {
         System.out.println("The script has started");
+        if (IS_SAVE_ENABLED) {
+            Context context = new Context();
+            try {
+                Process process = processService.find(context, processId);
+                processService.start(context, process);
```

Comment:
You are right. I removed it because I was thinking that if the process is fast, the user will not see the status `RUNNING` but only `COMPLETED`. But if it takes a lot of time, the user might see only `NULL` and `COMPLETED` as `process.status`. Added the `context.complete()`